### PR TITLE
K8s: Register apiserver as background service, and list real playlists

### PR DESCRIPTION
**Pull request by toddtreece**
**What is this feature?**

Registers `grafana-apiserver` with the background service registry. This currently seems like the simplest option while the handlers are tied to the existing SQL storage. The apiserver can be split into a separate target once unified storage is ready for production.

Required config changes:
```ini
app_mode = development

[feature_toggles]
grafanaAPIServer = true
```

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

@grafana/grafana-app-platform-squad 

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by ryantxu — APPROVED** [review]

File: pkg/services/grafana-apiserver/service.go [hunk 1]
```diff
@@ -76,26 +90,73 @@ type service struct {
...
+	rr       routing.RouteRegister
+	handler  web.Handler
+	builders []apis.APIGroupBuilder
 }
 
-func New(dataPath string) (*service, error) {
+func ProvideService(cfg *setting.Cfg,
+	rr routing.RouteRegister,
+	// Individual services -- these are actually added to the collection above
+	api0 *playlistv1.PlaylistAPIBuilder,
```
[the remark this answers is not in the document]
Comment (ryantxu) [reply pkg/services/grafana-apiserver/service.go:106]: I first added a layer that was a collection -- and each service (eg playlist) registered itself there, and then the collection was sent as a dependency to the apiserver.  We could do that, but it felt a bit equivalent but more convoluted 🤷🏻 -- when we are able to run a real decoupled service, that registration will have to change anyway.  I don't feel strongly on the best approach here.

File: public/locales/pseudo-LOCALE/grafana.json
Comment (joshhunt) [inline public/locales/pseudo-LOCALE/grafana.json:1]: if you like you can merge/rebase main to your PR to remove this from the diff.

File: pkg/services/grafana-apiserver/service.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (toddtreece) [reply pkg/services/grafana-apiserver/service.go:106]: @mildwonkey @ryantxu i'm still working out a bug, but here's the branch: https://github.com/grafana/grafana/pull/75389

File: pkg/apis/playlist/v1/handler.go [hunk 2]
```diff
@@ -47,54 +47,83 @@ func (r *handler) ConvertToTable(ctx context.Context, object runtime.Object, tab
...
 func (r *handler) List(ctx context.Context, options *internalversion.ListOptions) (runtime.Object, error) {
 	ns, ok := request.NamespaceFrom(ctx)
-	if ok && ns != "" {
-		orgId, err := apis.NamespaceToOrgID(ns)
-		if err != nil {
-			return nil, err
-		}
-		fmt.Printf("OrgID: %d\n", orgId)
+	if !ok || ns == "" {
+		return nil, fmt.Errorf("namespace required")
```
Comment (mildwonkey) [inline pkg/apis/playlist/v1/handler.go:51]: Not really related to this PR, but how does this work outside of cloud? do we use a generic namespace for single-tenant deployment?

Comment (ryantxu) [reply pkg/apis/playlist/v1/handler.go:51]: "default" is orgId=1, that seems easier than removing namespacing entirely for single-tenant deployments.  Essentially it adds "namespaces/default/" to a bunch of URLs when that is not really necessary (but makes the patterns consistent for single or multi-tenant)

Comment (ryantxu) [reply pkg/apis/playlist/v1/handler.go:51]: interesting, in that case we will have to check the orgs the user can see and update the query

**Pull request merged**